filebeat/module/elasticsearch/slowlog/_meta/fields.yml

```diff
@@ -10,14 +10,16 @@
   - name: took
     description: "Time it took to execute the query"
     example: "300ms"
-    type: text
+    # Should we extrac that nicer?
+    type: keyword
```

Comment:
It would definitely be nicer but the only way I can think of is by using the script processor with a bunch of conditions for various duration suffixes, something like:

https://github.com/elastic/beats/blob/2754572827a27241e4bdc71c37acf31780636473/filebeat/module/elasticsearch/server/ingest/pipeline.json#L36-L46

Is this what you had in mind?

Also, once we have the structured version of the slowlog, this field will (hopefully 🤞)  be in a standardized unit, so maybe in `master`/`7.0` we can call the field `took.text` if it's extracted from the unstructured slow log OR `took.ns` or something like that if it's extracted from the structured log?

Reply:
yes. It's odd that in the log this can contain different units. In any case I think it should be keyword for now?

Reply:
+1 to making this `keyword` and adding some script processing to parse out number and units into separate fields.

Reply:
Field dropped in https://github.com/elastic/beats/pull/10414.